src/docfx/docfx.csproj

```diff
@@ -7,6 +7,7 @@
     <CodeAnalysisRuleSet>../../stylecop.ruleset</CodeAnalysisRuleSet>
     <RootNamespace>Microsoft.Docs.Build</RootNamespace>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <TreatWarningsAsErrors Condition="'$(Configuration)'=='Release'">true</TreatWarningsAsErrors>
```

Comment:
>'$(Configuration)'=='Release' [](start = 38, length = 29)

suggest to enable for all, the local build can fix these problem before pushing to remote